features/step_definitions/functional_tests/functional_tests_steps.rb

```diff
@@ -30,7 +30,11 @@
 And(/^the authentication for the "([^"]*)" is established$/) do |package_name|
   json_keystone = proposal("keystone")
 
-  auth_node = json_keystone["deployment"]["keystone"]["elements"]["keystone-server"].first
+  if json_keystone["attributes"]["keystone"]["ha"]["enabled"] == true
+    auth_node = json_keystone["deployment"]["keystone"]["elements_expanded"]["keystone-server"].first
```

Comment:
Nearly sure this won't work, as you can't blindly connect to the usual port for HA -- it requires haproxy running on that node.

Isn't it possible to just parse .openrc instead? Otherwise, you need to generate the virtual hostname for this...